apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {
+  useState,
+  createContext,
+  useContext,
+  ReactNode,
+  SetStateAction,
+  Dispatch,
+} from 'react';
+import { Box, useTheme } from '@mui/material';
+
+type TabSwitcherContextType = {
+  setActiveTabID: Dispatch<SetStateAction<string>>;
+  activeTabID: string;
+};
+
+//the name of this context will be TabSwitcherContext
+const TabSwitcherContext = createContext<TabSwitcherContextType | null>(null);
+
+type TabProps = {
+  id: string;
+  children: ReactNode;
+};
+
+function Tab({ id, children }: TabProps) {
+  const context = useContext(TabSwitcherContext);
+  const { palette } = useTheme();
+  if (!context) {
+    console.error(
+      'TabSwitcherContext not found, did you forget to use TabSwitcherContextProvider?'
+    );
+    return null;
+  }
+
+  const { activeTabID, setActiveTabID } = context;
+
+  const isActive = id === activeTabID;
+  const tabColor = palette.primary.dark;
+
+  return (
+    <Box
+      onClick={() => setActiveTabID(id)}
+      sx={{
+        backgroundColor: isActive ? palette.primaryAction.hover : 'transparent',
+        borderRight: isActive ? `2px solid ${tabColor}` : 'none',
+        '& > div > span': {
+          color: isActive ? tabColor : palette.text.primary,
+        },
+        cursor: 'pointer',
+      }}
+    >
+      {children}
+    </Box>
+  );
+}
+
+type TabPanelProps = {
+  whenActive: string;
+  children: ReactNode;
+};
+
+function TabPanel({ whenActive, children }: TabPanelProps) {
+  const context = useContext(TabSwitcherContext);
+  if (!context) {
+    console.error(
+      'TabSwitcherContext not found, did you forget to use TabSwitcherContextProvider?'
+    );
```

Comment:
Same for this one. Do we need this `console.error`?

Reply:
I think is useful to have a specific error during development. By default React does not show a comprehensive error when component's provider is missing.